### PR TITLE
[Inductor][FlexAttention] Add helper functions of converting score_mod to block_mask

### PR DESCRIPTION
**Pull request by yanboliang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129909
* #129859
* #129831

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

File: torch/nn/attention/_flex_attention.py [hunk 1]
```diff
@@ -142,6 +142,71 @@ def _create_block_mask(
     ).as_tuple()
 
 
+def _create_mask(
```
Comment (drisspg) [inline torch/nn/attention/_flex_attention.py:145]: I think this is better, for my sanity the actual user facing API will be, 
create_block_mask(score_mod, b, h,m,n, device)? And return our tuple struct?

Comment (yanboliang) [reply torch/nn/attention/_flex_attention.py:145]: Right! I think ```create_block_mask``` would be the user facing API, we can also consider ```create_mask``` as public as well.

**Review by drisspg — APPROVED** [review]

**Pull request closed without merging**